### PR TITLE
add dependnecy type children: if class overwrite updates, namespace page should be rebuilt

### PR DESCRIPTION
**Pull request by ansyral**

### CODE REVIEW

**Comment by vwxyzh** [comment]
1. `FillReferenceInformation` only fill the children reference.
2. report children dependency to `FillReferenceInformation`.

File: src/Microsoft.DocAsCode.Build.ManagedReference/BuildManagedReferenceDocument.cs [hunk 1]
```diff
@@ -30,6 +31,13 @@ protected override void BuildArticle(IHostService host, FileModel model)
                     host.ReportDependencyTo(model, r.Uid, DependencyItemSourceType.Uid, DependencyTypeName.Reference);
                 }
             }
+
+            foreach (var child in from i in pageViewModel.Items
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.ManagedReference/BuildManagedReferenceDocument.cs]: need we add some test case?

File: src/Microsoft.DocAsCode.Build.ManagedReference/BuildManagedReferenceDocument.cs [hunk 2]
```diff
@@ -21,7 +22,7 @@ protected override void BuildArticle(IHostService host, FileModel model)
         {
             var pageViewModel = (PageViewModel)model.Content;
 
-            BuildArticleCore(host, model, shouldSkipMarkup:pageViewModel?.ShouldSkipMarkup ?? false);
+            BuildArticleCore(host, model, shouldSkipMarkup: pageViewModel?.ShouldSkipMarkup ?? false);
 
             foreach (var r in pageViewModel.References)
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.ManagedReference/BuildManagedReferenceDocument.cs]: >References [](start = 44, length = 10)

could we also move this logic into fillRefInfo?

Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.ManagedReference/BuildManagedReferenceDocument.cs]: fillRefInfo only fill reference for children

File: src/Microsoft.DocAsCode.Build.ManagedReference/BuildManagedReferenceDocument.cs [hunk 3]
```diff
@@ -5,6 +5,7 @@ namespace Microsoft.DocAsCode.Build.ManagedReference
 {
     using System.Collections.Generic;
     using System.Composition;
+    using System.Linq;
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.ManagedReference/BuildManagedReferenceDocument.cs]: remove

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by vwxyzh** [comment]
:shipit:

**Pull request merged**